### PR TITLE
Preapre codebase for FileRequest compiletime improvements

### PR DESCRIPTION
**Pull request by nemerle**
This is preparing grounds for the next PR that will contain the 'meat'
of the changes.
This is 90% whitespace cleanup + 10% namespace coalescing.

Signed-off-by: nemerle <96597+nemerle@users.noreply.github.com>

### CODE REVIEW

**Review by hultonha — APPROVED** [review]
Awesome! 😄 I highly recommend other reviewers look at this with 'show whitespace' set to false too 🙂

File: Code/Framework/AzFramework/AzFramework/IO/RemoteStorageDrive.h [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 namespace AzFramework
 {
+
```
[the remark this answers is not in the document]
Comment (nemerle) [reply Code/Framework/AzFramework/AzFramework/IO/RemoteStorageDrive.h]: Corrected in 0b5e7d9

[the remark this answers is not in the document]
Comment (hultonha) [reply Code/Framework/AzFramework/AzFramework/IO/RemoteStorageDrive.h]: Legend thanks 🙂 Just kicking a build off now 👍

**Review by amznestebanpapp — APPROVED** [review]

**Review by AMZN-koppersr — APPROVED** [review]

**Comment by AMZN-koppersr** [comment]
@nemerle I've also taken a quick look at PR 6092 to see what else is being planned. I'm a little concerned with those changes to be honest as I think there's a better way to reduce compile times without adding the rework. In particular I'm concerned about the addition of the Pimpl (-like) pattern. What I would prefer to see is this:
- Make `IStreamer.h` a true public interface without any implementation details. The biggest point that was never completed is removing the dependency on `FileRequestPtr`. This can just be a `void*` with a `reinterpret_cast` inside the implementation. This will also remove the need for `#include <AzCore/IO/Streamer/FileRequest.h>` which will make for a smaller change and more targeted change that suggested in PR 6092.
- Move the implementation of AZ::IO::Streamer into a gem. This will reduce the overall size of AzCore which will decrease link times on non-monolithic platforms and improves turn around times for future development of AZ::IO::Streamer. It will also allow users to use a different file system implementation as not everybody will agree with the design choices in AZ::IO::Streamer.

Besides time, there are a few reasons why the above changes haven't happened yet:
- There's no clear pattern to follow with regards to public/private APIs and how to version them. Is it better to have every call go through `AZ::Interface` or have a base class and have a limited `AZ::Interface` with just a create/destroy that works with the base class? What does the versioning mechanic look like? Is `void*` an acceptable way to decouple implementations or does there need to an approach with stronger type safety?
- What will a AzCore/AzFramework gem look like? Should it all be separate modules/targets or will that eventually cause an excessive amount of binaries? Is that even an issue to be concerned about now or can it be tackled when/if it becomes an issue? What else can be moved to this foundational gem(s)?

With topics like modularization and API stability in mind the above questions and others like that should get an answer imho, but in the meantime at the very least `FileRequestPtr` in `IStreamer.h` can be turned into a type-agnostic define like `using FileRequest = void*` and have all the implementation details moved to AZ::IO::Streamer itself.

**Comment by nemerle** [comment]
@AMZN-koppersr I'm unsure if replacing `FileRequestPtr` with a `void*` is the best approach to decoupling things, especially since it's currently a smart pointer?

**Comment by AMZN-koppersr** [comment]
> @AMZN-koppersr I'm unsure if replacing `FileRequestPtr` with a `void*` is the best approach to decoupling things, especially since it's currently a smart pointer?

It'll require some updates. (Possibly some form of a "smart void pointer".) At it's core though the current `FileRequestPtr` is an implementation detail of AZ::IO::Streamer and imho shouldn't be part of the public API, especially if it's going to be turned into a gem at some point. As a side effect, such a decoupling should also fix the compile overhead that `FileRequestPtr` spreads throughout the code.

I'm tentatively scheduled to do a pass on the scheduler for AZ::IO::Streamer early next year so if you like I can also make the `FileRequestPtr` update then?

**Comment by nemerle** [comment]
I think bundling FileRequestPtr change with 'gemification' of the Streamer makes sense :smile:

File: Code/Framework/AzFramework/AzFramework/IO/RemoteStorageDrive.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (nemerle) [reply Code/Framework/AzFramework/AzFramework/IO/RemoteStorageDrive.h]: The build fails on Linux, caused by asset processor timeout. Should I try merging current development to see if it fixes anything?

**Comment by hultonha** [comment]
@nemerle I think you need to update the branch to latest again and then I can kick off another build, should be good to merge then 👍

**Comment by nemerle** [comment]
@hultonha I'll do a merge as soon as I'm able

**Comment by hultonha** [comment]
Awesome thanks @nemerle 👍

**Comment by nemerle** [comment]
@hultonha - current development merged in, had to do it twice :p

**Comment by hultonha** [comment]
Nice one thanks @nemerle, kicking off a build now :+1:

**Comment by hultonha** [comment]
Merged! 😄

**Pull request merged**